captura/views.py

```diff
@@ -224,7 +227,153 @@ def create_estudio(request):
+        if integrante_form.is_valid():
+            integrante_form.save()
+            rol = integrante_form.cleaned_data['Rol']
+            if rol == integrante_form.OPCION_ROL_ALUMNO:
+                alumnos = Alumno.objects.filter(integrante=integrante)
+                if alumnos:
+                    alumno = alumnos[0]
+                    alumno_form = AlumnoForm(request.POST, instance=alumno)
+                    if alumno_form.is_valid():
+                        alumno_form.save()
+                else:
+                    escuela = Escuela.objects.all().first()
+                    Alumno.objects.create(integrante=integrante, escuela=escuela)
+                    return redirect(reverse('captura:integrante',
+                                            kwargs={'id_integrante': id_integrante}))
+
+            elif rol == integrante_form.OPCION_ROL_TUTOR:
+                tutores = Tutor.objects.filter(integrante=integrante)
+                if tutores:
+                    tutor = tutores[0]
+                    tutor_form = TutorForm(request.POST, instance=tutor)
+                    if tutor_form.is_valid():
+                        tutor_form.save()
+                else:
+                    Tutor.objects.create(relacion=integrante_form.cleaned_data['Rol'],
+                                         integrante=integrante)
+                    return redirect(reverse('captura:integrante',
+                                            kwargs={'id_integrante': id_integrante}))
+
+            return redirect(reverse('captura:integrantes',
+                                    kwargs={'id_familia': integrante_form.instance.familia.pk}))
+    forms = {}
+    integrante = Integrante.objects.get(pk=id_integrante)
+    rol_integrante = 'ninguno'
+    rol_disabled = False
+
+    alumnos = Alumno.objects.filter(integrante=integrante)
+    if alumnos:
+        alumno = alumnos[0]
+        forms['form_alumno'] = AlumnoForm(instance=alumno)
+        rol_integrante = 'alumno'
+        rol_disabled = True
+
+    tutores = Tutor.objects.filter(integrante=integrante)
+    if tutores:
+        tutor = tutores[0]
+        forms['form_tutor'] = TutorForm(instance=tutor)
+        rol_integrante = 'tutor'
+        rol_disabled = True
+
+    if integrante_form:
+        forms['integrante_form'] = integrante_form
+    else:
+        forms['integrante_form'] = IntegranteForm(instance=integrante,
+                                                  initial={'Rol': rol_integrante})
+    if rol_disabled:
+        forms['integrante_form'].fields['Rol'].widget = HiddenInput()
+    form_view = 'captura/integrante_form.html'
+    return render(request, 'captura/captura_base.html', {'form_view': form_view,
+                                                         'forms': forms,
+                                                         'integrante': integrante})
+
+
+@login_required
+@user_passes_test(is_capturista)
+def create_integrante(request, id_familia):
```

Comment:
Ok now I'm just understanding the form you included and I asked about (the submit form instead of a href link).

Now my question is why are we doing this in the first place? I mean, the form should be rendered blank when clicking "crear estudio" and then, creating the user if the form is valid don't you think? 

I don't like this because if after clicking agregar integrante I don't fill the form entirely the user is already created and there's no way to delete it (and no, I think just adding a delete button is not the wisest idea). That's what I meant when I said that there should be a view for creating and another for editing it. I don't understand why this wasn't the default choice in the first place.